tools/inference.py

```diff
@@ -108,65 +171,284 @@ def make_model(**kwargs):
+                if scale_features == "median_std":
+                    features[feature] = (features[feature] - stats["median"]) / stats[
+                        "std"
+                    ]
+                elif scale_features == "min_max":
+                    features[feature] = (features[feature] - stats["min"]) / (
+                        stats["max"] - stats["min"]
+                    )
+        te = time.time()
+        if tm:
+            print(
+                "min max scaling".ljust(JUST)
+                + "\t --> \t"
+                + str(round(te - ts, 4))
+                + " s"
+            )
 
-    # scale features
-    train_config = config["training"]["classes"][model_class]
-    feature_names = config["features"][train_config["features"]]
-    feature_stats = config.get("feature_stats", None)
-    scale_features = "min_max"
-
-    for feature in feature_names:
-        stats = feature_stats.get(feature)
-        if (stats is not None) and (stats["std"] != 0):
-            if scale_features == "median_std":
-                features[feature] = (features[feature] - stats["median"]) / stats["std"]
-            elif scale_features == "min_max":
-                features[feature] = (features[feature] - stats["min"]) / (
-                    stats["max"] - stats["min"]
-                )
+        if verbose:
+            print(features)
+
+        # Load pre-trained model
+        ts = time.time()
+        if str(path_model).endswith(".h5"):
+            model = tf.keras.models.load_model(path_model)
+        else:
+            model = make_model(features_input_shape=(len(feature_names),))
+            if verbose:
+                print(model.summary())
+            model.load_weights(path_model).expect_partial()
+        te = time.time()
+        if tm:
+            print(
+                "load pre-trained model".ljust(JUST)
+                + "\t --> \t"
+                + str(round(te - ts, 4))
+                + " s"
+            )
+
+        # Impute missing data and flag source ids containing missing values
+        make_missing_dict(source_ids)
+        features = clean_data(
+            features,
+            feature_names,
+            feature_stats,
+            field,
+            ccd,
+            quad,
+            flag_ids,
+            whole_field,
+        )
+
+        ts = time.time()
+        preds = model.predict([features[feature_names].values, dmdt])
+        features[model_class + '_dnn'] = preds
+        te = time.time()
+        if tm:
+            print(
+                "dnn inference (model.predict())".ljust(JUST)
+                + "\t --> \t"
+                + str(round(te - ts, 4))
+                + " s"
+            )
+        preds_df = features[["_id", model_class + '_dnn']].round(2)
+        preds_df.reset_index(inplace=True, drop=True)
+    else:
+        # xgboost inferencing
+        train_config = path_model[-7]
+        feature_names = config["inference"]["xgb"][train_config]
+
```

Comment:
Should clean_dataset_xgb be called here to prepare the features?

Reply:
I did not keep it then because all the features that contained missing values were not inputs to the xgb models. However, I now realize it is better to keep a safety clean_dataset function to prepare the features in case of future updates to either the model inputs or the dataset. Should I add it now?

Reply:
Sounds good, probably best to add it for the future like you mentioned.